src/code/audio/code_801A5BD0.c

```diff
@@ -1,45 +1,835 @@
+    }
+
+    entryIndex = gSfxBanks[bankId][0].next;
+    k = 0;
+    while (entryIndex != 0xFF) {
+        if ((gSfxBanks[bankId][entryIndex].state == SFX_STATE_QUEUED) &&
+            (gSfxBanks[bankId][entryIndex].sfxId & 0xC00)) {
+            gSfxBanks[bankId][entryIndex].freshness--;
+        } else if (!(gSfxBanks[bankId][entryIndex].sfxId & 0xC00) &&
+                   (gSfxBanks[bankId][entryIndex].state == SFX_STATE_PLAYING_2)) {
+            Audio_QueueCmdS8((gSfxBanks[bankId][entryIndex].channelIdx << 8) | 0x6020000, 0);
+            Audio_RemoveSfxBankEntry(bankId, entryIndex);
+        }
+        if (gSfxBanks[bankId][entryIndex].freshness == 0) {
+            Audio_RemoveSfxBankEntry(bankId, entryIndex);
+        } else if (gSfxBanks[bankId][entryIndex].state != SFX_STATE_EMPTY) {
+            entry = &gSfxBanks[bankId][entryIndex];
+
+            if (&gSfxDefaultPos.x == entry[0].posX) {
+                entry->dist = 0.0f;
+            } else {
+                entryPosY = *entry->posY * 1;
+                entryPosX = *entry->posX * 0.5f;
+                entry->dist = (SQ(entryPosX) + SQ(entryPosY) + SQ(*entry->posZ)) / 10.0f;
+            }
+
+            sfxImportance = entry->sfxImportance;
+
+            if (entry->sfxParams & 0x10) {
+                entry->priority = SQ(0xFF - sfxImportance) * SQ(76);
+            } else {
+                if (entry->dist > 0x7FFFFFD0) {
+                    entry->dist = 0x70000008;
+                }
+                entry->priority = (u32)entry->dist + (SQ(0xFF - sfxImportance) * SQ(76));
+                if (*entry->posZ < 0.0f) {
+                    entry->priority += (s32)(-*entry->posZ * 6.0f);
+                }
+            }
+
+            if (entry->dist > SQ(1e5f)) {
+                if (entry->state == SFX_STATE_PLAYING_1) {
+                    Audio_QueueCmdS8((entry->channelIdx << 8) | 0x6020000, 0);
+                    if (entry->sfxId & 0xC00) {
+                        Audio_RemoveSfxBankEntry(bankId, entryIndex);
+                        entryIndex = k;
+                    }
+                }
+            } else {
+                numChannels = gChannelsPerBank[gSfxChannelLayout][bankId];
+
+                for (i = 0; i < numChannels; i++) {
+                    if (chosenSfxs[i].priority >= entry->priority) {
+                        if (numChosenSfxs < gChannelsPerBank[gSfxChannelLayout][bankId]) {
+                            numChosenSfxs++;
+                        }
+
+                        for (j = numChannels - 1; j > i; j--) {
+                            chosenSfxs[j].priority = chosenSfxs[j - 1].priority;
+                            chosenSfxs[j].entryIndex = chosenSfxs[j - 1].entryIndex;
+                        }
+
+                        chosenSfxs[i].priority = entry->priority;
+                        chosenSfxs[i].entryIndex = entryIndex;
+                        i = numChannels; // "break; i.e. force for-loop to end"
```

Comment:
How about
```suggestion
                        i = numChannels; // make this last pass through loop, here equivalent to "break"
```
mainly because this is *not* equivalent to break if you add stuff later on.